### PR TITLE
Extend the use of characteristics

### LINKED ISSUE #94: Reconsidering characteristics
Remove Characteristic Input Types
Implement OlmCharacteristic as a separate class and replace the previous string in LineInput
Implement qCharacteristic as a separate class and replace the previous string in the corresponding input

Replace existing characteristics available as strings with classes

### PR DESCRIPTION
**Pull request by ckittl**
Closes #94

### CODE REVIEW

**Comment by ckittl** [comment]
!test

**Comment by ckittl** [comment]
The apparent Codacy issues are "old" ones.

File: src/main/java/edu/ie3/datamodel/io/factory/input/LineInputFactory.java [hunk 1]
```diff
@@ -53,10 +55,19 @@ protected LineInput buildModel(
                         ArrayUtils.addAll(
                             NodeInput.DEFAULT_GEO_POSITION.getCoordinates(),
                             NodeInput.DEFAULT_GEO_POSITION.getCoordinates())));
-    final Optional<String> olmCharacteristic =
-        data.containsKey(OLM_CHARACTERISTIC)
-            ? Optional.of(data.getField(OLM_CHARACTERISTIC))
-            : Optional.empty();
+    final OlmCharacteristicInput olmCharacteristic;
+    try {
+      olmCharacteristic =
```
Comment (johanneshiry) [inline src/main/java/edu/ie3/datamodel/io/factory/input/LineInputFactory.java:60]: good implementation! 👍

File: src/main/java/edu/ie3/datamodel/io/processor/EntityProcessor.java [hunk 2]
```diff
@@ -258,9 +259,15 @@ private String processMethodResult(Object methodReturnObject, Method method, Str
       case "HpTypeInput":
         resultStringBuilder.append(((UniqueEntity) methodReturnObject).getUuid());
         break;
-      case "Optional": // todo needs to be removed asap as this is very dangerous, but necessary as
```
Comment (johanneshiry) [inline src/main/java/edu/ie3/datamodel/io/processor/EntityProcessor.java:261]: 🎉🎉

File: src/main/java/edu/ie3/datamodel/io/processor/EntityProcessor.java [hunk 3]
```diff
@@ -258,9 +259,15 @@ private String processMethodResult(Object methodReturnObject, Method method, Str
...
-        resultStringBuilder.append(((Optional<String>) methodReturnObject).orElse(""));
+      case "EvCharacteristicInput":
+      case "OlmCharacteristicInput":
+      case "WecCharacteristicInput":
+      case "CosPhiFixed":
+      case "CosPhiP":
+      case "QV":
+      case "ReactivePowerCharacteristic":
+      case "CharacteristicInput":
+        resultStringBuilder.append(((CharacteristicInput) methodReturnObject).deSerialize());
```
Comment (johanneshiry) [inline src/main/java/edu/ie3/datamodel/io/processor/EntityProcessor.java:270]: Dunno if this makes sense/works, but it turns out raw unparametrized complains 😅 

```suggestion
        resultStringBuilder.append(((CharacteristicInput<?,?>) methodReturnObject).deSerialize());
```

File: src/main/java/edu/ie3/datamodel/io/factory/typeinput/SystemParticipantTypeInputFactory.java [hunk 4]
```diff
@@ -186,7 +191,19 @@ private SystemParticipantTypeInput buildWecTypeInput(
 
     ComparableQuantity<Length> hubHeight = data.getQuantity(HUB_HEIGHT, StandardUnits.HUB_HEIGHT);
 
-    return new WecTypeInput(uuid, id, capEx, opEx, cosPhi, etaConv, sRated, rotorArea, hubHeight);
+    WecCharacteristicInput cpCharacteristic = null;
```
Comment (johanneshiry) [inline src/main/java/edu/ie3/datamodel/io/factory/typeinput/SystemParticipantTypeInputFactory.java:194]:
```suggestion
    WecCharacteristicInput cpCharacteristic;
```

File: src/test/groovy/edu/ie3/datamodel/io/factory/input/LineInputFactoryTest.groovy [hunk 5]
```diff
@@ -36,7 +43,56 @@ class LineInputFactoryTest extends Specification implements FactoryTestHelper {
...
+			assert !operationTime.endDate.present
+			assert operator == operatorInput
+			assert id == parameter["id"]
+			assert nodeA == nodeInputA
+			assert nodeB == nodeInputB
+			assert type == typeInput
+			assert noOfParallelDevices == Integer.parseInt(parameter["paralleldevices"])
+			assert length == getQuant(parameter["length"], StandardUnits.LINE_LENGTH)
+			assert geoPosition == getGeometry(parameter["geoposition"])
+			olmCharacteristic.with {
```
Comment (johanneshiry) [inline src/test/groovy/edu/ie3/datamodel/io/factory/input/LineInputFactoryTest.groovy:73]: 👍

**Pull request merged**